Krav Maga: Stompies Edition

necromanceranne opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

An alternative to https://github.com/tgstation/tgstation/pull/49079 and https://github.com/tgstation/tgstation/pull/49099  ~~because that changes the availability of krav maga and not the martial art itself. I do not consider this to be related to that change and neither do I think this change is invalidated by what ATHATH is attempting to do.~~ because I noticed he's actually changing more than availability, but also the actual abilities of krav maga. Eurgh.

This makes legsweep do a fairly decent amount of stamina damage and long knockdown. In addition, disarm intent **is now a nonlethal jab, which do more stamina damage on targets who are on the floor**. They also respect melee armor. **They also have a chance to disarm based on stamina loss, which makes them much stronger against people with high stamina damage.**

~~Come to think of it, I don't even think harm stomps respect armor. ~~

~~Come to think of it I don't think any martial art even does for the most part. But most are antag shit so I'm not gonna change that.~~

I've made krav maga's harm intent respect melee armor and additionally have made it more variable than 10 punch damage and 15 stomp.

The attack description is subject to change. I have idea how to best describe it.

Also I'm not a very good coder so if I did something wrong please yell at me. It compiled okay? It didn't even runtime!

## Why It's Good For The Game

I do agree that krav maga really needs to have the stun stripped out but I don't want to entirely remove the bite from the martial art and I want it to be an asset to the person who will be using it. The warden. This way, it's actually still useful for all the reasons you'll be using it! VIOLENT COMPLIANCE.

Closes #49079 

## Changelog
:cl:
balance: Removes the stun from Krav Maga and makes it a high duration knockdown with stamina damage (20-30), respecting chest armor for the damage.
adds: Disarm intent is now a nonlethal jab that does (5-10) stamina damage. If used on people who are prone, it does (10-15) stamina damage. It also has a chance to completely disarm someone based on the amount of stamina damage they have.
balance: Krav Maga harm stomps/punches now respect armor and have a variance of (5-10) for punches and (5-10+5) for stomps.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

necromanceranne committed:
Changes how krav maga works by making the legsweep a knockdown with stamina damage and disarm intent on someone on the floor a nonlethal strike. These hopefully respect armor. Needs further adjustments most likely

HomocidlePancakes commented:
It’s still better to just take the gloves off and to push.

necromanceranne committed:
Made harm stomp respect armor as well because 15 force kicks with no mitigation is fucking stupid and really made nonlethal stomps look like crap

necromanceranne commented:
@HomocidlePancakes that's a problem with shoves being a stun and not with krav maga. A telebaton is also better than legsweep for much the same reasons.

HomocidlePancakes commented:
Don't see anything wrong with shoves being able to stun. You first need to knock the lad, and avoiding being pushed in to a wall is painfully easy. You won't be able to pull the stun off w/o some sort of a outside tool.

But let's ignore all that for a moment. 5-15 stamina damage. That's not much at all. You'll end up making this as viable as old pepperspray.

necromanceranne commented:
Okay well again that's not really at all the point of the change, and again, outside the scope of the balance change. If the stamina damage was any more than what I set, you would stamcrit people at lightspeed and it would as though nothing has changed. I had the values much higher originally but when I tested it, it was obscenely fast for my taste. 

It's not actually that slow right now, either. Especially when you have a baton and have a knockdown with a stamina hit on top of that. All I want to do is broaden the abilities of krav maga without reliance on hard stun mechanics. I won't add floor shoves because of this. I might add additional mechanics relying on the target being on the ground and remove the disarm. So you can shove people to the floor, if you want, ignoring your conditionless knockdown with damage.

HomocidlePancakes commented:
This pointless. I'll just say that whatever you end up doing shouldn't make the user take off gloves to do a shovestun or make them switch to a baton because that's a better option in all cases. Also, don't forget about ops.

trollbreeder commented:
Please stop with the krav maga PR arms race.

Shadowflame909 commented:
I like these stamina gloves. Could do with a bit more power creep on the stamina part though. Gotta make em more incentivising then a Stun-Baton.

wesoda25 commented:
10 second knockdown is way too long, especially if wardens can stack more stamina damage on top of it. Make it like 5 or something

Shadowflame909 commented:
Can't you still assault people while knocked down. It just prevents running if I recall.

anconfuzedrock commented:
why is this pr the good one

AtlasRules commented:
as long as i can do the **john kayloo special** with krav maga im happy

ATH1909 commented:
I think if every krav maga disarm performed on a knocked down person made them drop what they're holding (so it's harder for your victim to just whip out a .357 and ice you), leg sweep->disarm spam would be a lot more viable.

PKPenguin321 commented:
for the record the original intent of legsweep was a VERY brief stun to put a dead stop runners for a second while also not being enough to cuff or anything on its own. would also work as an effective disarm. ideally you'd use it on them then would be able to switch to a stun baton if you were really fast or have another officer with a baton at the ready.
i dont think making it do knockdown is worthwhile. the dude on the floor can just pick his item back up and/or continue moving away from you and it ruins the whole thing. what's the purpose of it now? it doesn't really seem to have one.

necromanceranne commented:
@ATH1909 originally when I had it disarming, it felt like too much and I didn't feel like it was good balance wise to completely strip any opportunities from the person being beat from responding to this dude with baton hands. I kept it to just stamina damage and I'll add some more disarm opportunities elsewhere.

@PKPenguin321 games significantly evolved since krav's original implementation. 4 seconds isn't enough to cuff but it damn well is enough to kill the fucker you legsweeped, and click speed isn't a factor when you can line stun batons up on a shift+e hotkey to instantly baton people down. My literal goto is sweep shift+e baton and they're completely fucking helpless for any followup I had. This includes the compact combat shotgun, which kills obscenely fast. The length of the knockdown is meant to make it more of a slowdown to open a broader range of options for the user to disable someone with. Back off and shoot, stun baton them, and failing that, they have stamina stomps. My intent (hopefully, barring shitty implementation on my part) is to add layering disabling effects that climb people up towards becoming completely helpless, while also being complimentary to security equipment and not overshadowing too it, since at present that is a very much the case.

My intent with the stomps is that they're not supposed to completely replace stun batons. They're supposed to give you supplementary tool to your arsenal by allowing you to detain people even when disarmed of weapons. You could kill people very terrifyingly quickly with krav maga but it seemed weird how the martial art wasn't able to help you control people's actions more and disable them in a way that didn't bring them close to death, given who was supposed to be using the gloves.

ATH1909 commented:
> @ATH1909 originally when I had it disarming, it felt like too much and I didn't feel like it was good balance wise to completely strip any opportunities from the person being beat from responding to this dude with baton hands. I kept it to just stamina damage and I'll add some more disarm opportunities elsewhere.

I mean, the victim still *can* do things; they just have to time them so that they pull out their item and use it in-between disarm shoves.

Leaving the disarm in this PR as it is currently is gonna cause wardens to leg sweep people, then take off their gloves and disarm them for a stun that's long enough to cuff. A martial art's disarm_act() (or harm_act() or grab_act() or whatever) _shouldn't be worse than a normal disarm_act()_ (unless it's something like the mushpunch martial art, which is kind of intentionally bad).

> My intent with the stomps is that they're not supposed to completely replace stun batons. They're supposed to give you supplementary tool to your arsenal by allowing you to detain people even when disarmed of weapons.

The thing is, when you get disarmed of your baton as a warden, you're not gonna try to use this new!Krav Maga, you're gonna either back up and use your disabler or take out your spare baton. Krav Maga _should_ be comparable in strength to a stun baton, or else wardens are just gonna shift+e->left click instead of ever using their Krav Maga.

necromanceranne committed:
Krav Maga disarm intent is now a nonlethal jab. If jabbed, you have a probability based on stamina damage to drop everything you're holding. Krav stomp slightly more variable in damage to match nonlethal jab.

necromanceranne commented:
Added some changes:
1. Tightened the stamina damage for floor stomps and made disarms inherently a nonlethal jab that does less damage.
2. These nonlethal jabs can disarm based on stamina loss.
3. Removed the horrible disarm completely.
4. Balanced up the harm stomps a bit.

PKPenguin321 commented:
If that's the case I would keep the stun but half the duration and remove the damage.

necromanceranne commented:
Well stuns are being phased from the game and this is an outlier. Whether it's two seconds or four, it'll still stand at odds a bit with security gear, and even two seconds seems fairly strong. So I'm giving it an overhaul to hopefully make it more compliant with security equipment in general, and hopefully internally consistent and at least somewhat consistent with standard unarmed attacks. 

Legsweep fills the role of shove in all but the shovestun (which is a mechanic that may or may not get phased out, iunno what the deal is with that and again not worth considering for the sake of this change due to it being tentatively retained in the game), and the disarm just didn't really make much sense to keep when we had ditched RNG disarms a reasonably long time ago, and legsweep served that purpose so much better. Even shove itself served the purpose better, but I wanted to keep legsweep. Consistent and dependable mechanics are better overall, but I also want to give opportunities for the prepared to protect themselves.

All in all, it'll be a much more internally consistent and usable martial art. Hopefully.

PKPenguin321 commented:
>Well stuns are being phased from the game and this is an outlier. Whether it's two seconds or four, it'll still stand at odds a bit with security gear, and even tWo SeCoNdS sEeMs FaIrLy StRoNg.

no, it does not. actually to be honest i think the new leg sweep is even stronger, at least for murdering people, because you more than doubled the knockdown on it. i just don't think that that's what it should be good for. i guess if you've got this mindset of "stuns bad! 2 second stun is stronger than 10 second knockdown!" then there's no way i can change your mind though, because to believe something so blatantly untrue means you're too far gone.

also the new disarm is really meh. just FYI because I rewrote normal punches to do stamina, what you've got here will be mostly outhealed, you may want to test. i would at the absolute minimum make the disarm punch do 1.5x as much damage as you've got here. that said, i'm not sure it even needs changing.. what's so bad about it being able to grab items away?

necromanceranne commented:
Stun batons are also 10 seconds, and do far more stamina damage. Telebatons have moderate stamina damage and very low knockdowns. So, in terms of power scale, this is somewhere in the middle. If you could kill someone with legsweep, what was really stopping you from doing it with a baton other than one of this could be disarmed from you? And isn't that part of the point of martial arts?

There is also a big mechanical difference to stuns and knockdowns. Two seconds of helplessness means you straight up can't use most abilities and items where as a knockdown does allow you to use some abilities and most items, including pulling weapons of your own. I've seen enough edge cases recently to notice the difference. Knockdowns in of themselves are powerful but they do not completely strip you of choice. 

Also stamina natural regen hasn't been a thing for almost a year. https://github.com/tgstation/tgstation/pull/43966

PKPenguin321 commented:
>Also stamina natural regen hasn't been a thing for almost a year.

Forgot about that, good point, nvm then

Anyways I don't agree that 10 seconds of knockdown is weaker than a 2 second stun. 2 seconds isn't very much at all, it's practically nothing. If you'll recall when slipping stunned, a banana peel that had 10~ potency would basically make you fall over and be up a moment later. That's the shortness of the stun we're talking about and there's not a chance that that's somehow more of a disadvantage than being forced to crawl around for a whole 10 seconds. I'm well aware of the mechanical differences between stuns and knockdowns and it's precisely why I bring this up. I think giving the knockdown a literal 2.5x duration is a ridiculous buff.

necromanceranne commented:
I suppose I'll give it a bit more thought. It's not like the warden is lacking for legsweeps...

I do need to consider whether shorter knockdowns warrant lower stamina damage on the sweep just so you don't have a situation where you do more stamina damage off cycling legsweeps than stomps. Probably not?

SkeletalElite commented:
Does this apply all the stamina damage to the chest or does it divide it out across the body? If it divides it out across the body I have some concerns that the stamina damage may be a little too low.

necromanceranne commented:
It's applied to the limb targeted and I'm fairly sure in legsweeps case it will always hit the chest.

SkeletalElite commented:
All I know is that stamina damage to arms and legs x.75 for the purpose of stamina crit, that possibly has something to do with it.

cacogen commented:
>Stun batons are also 10 seconds

no, they're five seconds

ATH1909 commented on code/datums/martial/krav_maga.dm:
```diff
@@ -130,12 +132,14 @@
 		bonus_damage += 5
 		picked_hit_type = "stomp"
-	D.apply_damage(bonus_damage, A.dna.species.attack_type)
+	D.apply_damage(rand(5,10) + bonus_damage, A.dna.species.attack_type, affecting, armor_block)
```
You do know that this doesn't factor in a species' base punch damage, right? You might wanna incorporate species.punchdamagelow in here somewhere.

trollbreeder replied:
Implying that golems are gonna know Krav Maga.

necromanceranne replied:
And also this is already partially how the damage works on krav maga already. The other main martial arts also do not use species punch damage. The only thing it carries is damage type for the sake of ethereals and I'm pretty sure thats it.

The way it worked before was that krav maga would do 10 damage on normal punch and 15 on kick because the bonus damage value was the only thing used in apply_damage. All I added was randomized values and retained the shape of the code.

I'm not really interested in breaking the mold here. Human punchdamagelow is like 1 which is dreadful anyway.

ATH1909 commented on code/datums/martial/krav_maga.dm:
```diff
@@ -151,22 +155,27 @@
+		log_combat(A, D, "stomped nonlethally")
+	if(prob(D.getStaminaLoss()))
+		D.visible_message("<span class='warning'>[D] sputters and recoils in pain!</span>", "<span class='userdanger'>You recoil in pain as you are jabbed in a nerve!</span>")
+		D.drop_all_held_items()
```
can you make it so that if this causes someone to drop an item, the game tries to put it in the warden's off hand before trying to put it on the ground?

necromanceranne replied:
This force drops everything so that's going to get messy and likely put items that don't fit under the warden as opposed to under the victim. It's not a rework of the disarm it's just a less severe version of how some martial arts stun you on damage thresholds.

ATH1909 commented on code/datums/martial/krav_maga.dm:
```diff
@@ -151,22 +155,27 @@
+		to_chat(A, "<span class='danger'>You jab [D]!</span>")
+		A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+		playsound(D, 'sound/effects/hit_punch.ogg', 50, TRUE, -1)
+		D.apply_damage(rand(5,10), STAMINA, affecting, armor_block)
```
you might want to make this disarm_act also apply shove slowdown (normal shoves slow people who(m) they hit down for a bit).

necromanceranne replied:
You already have legsweep. Just legsweep. I would rather it have parity with your standard punch over trying to incorporate too much of the shove power back into what is literally a decent stamina punch. Given stamina damage already reduces your run speed, having it apply a slow in of itself would make it obscenely more powerful than even the harm stomps for sticking on people when I want that to be what legsweep enables.

The reason why I made it a nonlethal strike is because legsweep does what shove does for everyone else.

necromanceranne committed:
Reduced the knockdown to 6 from 10.

ExcessiveUseOfCobblestone commented:
alternatives need to be formatted so it's `closes #PRnumber` else if the PR gets merged it will not autoclose the others.

necromanceranne commented:
stop bulli, bot, it's finished QwQ
